### DISCUSSION TITLE
SOC LAB4-2  .c的x跟tap 值 怎麼放進去

### DISCUSSION THREAD

**Discussion by M11107004** [bol-edu/HLS-SOC-Discussions#135] [Q&A]
因為tap地址值要累加放進去，想請問要怎麼把值放進去與硬體溝通？

**Comment by LingChiYang** [comment]
請參考
https://github.com/bol-edu/caravel-soc_fpga-lab/blob/main/lab-caravel_fir/lab4-2-caravel-testbench.pdf

(1) 先請同學確認firmware寫入地址是否生效，可用波型看wishbone相關信號。
(2) 若wishbone有成功傳遞位置與資料，請同學確認protocol是否正確。
(3) 以上兩步都完成，確認是否在程式的最後有傳出正確答案，若不是正確答案或是沒有看到endmark，請同學確認每一次的request有沒有符合以上兩步
